### PR TITLE
Build-Scan v2

### PR DESCRIPTION
**Pull request by sverdlov93**
- [ ] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----
New build scan API that works directly with Xray.
New docker scan API.

### CODE REVIEW

File: xray/services/buildscan.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	requestUrl := fmt.Sprintf("%s%s/%s/%s", bs.XrayDetails.GetUrl(), buildScanAPI, params.BuildName, params.BuildNumber)
+	if params.Project != "" {
+		requestUrl += projectKeyQueryParam + params.Project
+	}
+	syncMessage := fmt.Sprintf("Sync: Get Build Scan results. Build:%s/%s...", params.BuildName, params.BuildNumber)
+	httpClientsDetails := bs.XrayDetails.CreateHttpClientDetails()
+	utils.SetContentType("application/json", &httpClientsDetails.Headers)
+
+	//The scan request may take some time to complete. We expect to receive a 202 response, until the completion.
+	ticker := time.NewTicker(defaultSyncSleepInterval)
```
Comment (eyalbe4) [inline xray/services/buildscan.go:74]: It looks to me that this waiting mechanis, with the ticker is more compex that it should be.
Why don't we just have a loop with a fixed number of iterations (which is the wait timeout actually) and a 5 seconds sleep inside it. We also don't need to spawn a new go routine for it.

File: xray/services/summary.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	XrayDetails auth.ServiceDetails
+}
+
+// NewSummaryService creates a new service to retrieve the version of Xray
+func NewSummaryService(client *jfroghttpclient.JfrogHttpClient) *SummaryService {
+	return &SummaryService{client: client}
+}
+
+// GetXrayDetails returns the Xray details
+func (ss *SummaryService) GetXrayDetails() auth.ServiceDetails {
```
Comment (eyalbe4) [inline xray/services/summary.go:33]: This method seem to be unused.

File: xray/services/summary.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	if err != nil {
+		return nil, errorutils.CheckError(err)
+	}
+	if summaryResponse.Errors != nil && len(summaryResponse.Errors) > 0 {
+		return nil, errorutils.CheckErrorf("Getting build-summery for build: %s failed with error: %s", summaryResponse.Errors[0].Identifier, summaryResponse.Errors[0].Error)
+	}
+	return &summaryResponse, nil
+}
+
+func (ss *SummaryService) GetArtifactSummary(artifactSummaryRequest ArtifactSummaryRequest) (*SummaryResponse, error) {
```
Comment (eyalbe4) [inline xray/services/summary.go:61]: I suggest we remove this API, becuase it's unclear when it should be used.
Let's also remove the structs this API uses.

File: xray/manager.go [hunk 4]
```diff
@@ -132,6 +132,36 @@ func (sm *XrayServicesManager) GetScanGraphResults(scanID string, includeVulnera
 	return scanService.GetScanGraphResults(scanID, includeVulnerabilities, includeLicenses)
 }
 
+// BuildScan will send Xray the given build for scan
```
Comment (eyalbe4) [inline xray/manager.go:135]: // BuildScan will send Xray the given build for scan
-->
// BuildScan scans a published build-info with Xray.

File: xray/manager.go [hunk 5]
```diff
@@ -132,6 +132,36 @@ func (sm *XrayServicesManager) GetScanGraphResults(scanID string, includeVulnera
...
+
+// GetBuildScanResults returns an Xray build scan output of the requested build scan.
+// The scanId input should be received from BuildScan request.
+func (sm *XrayServicesManager) GetBuildScanResults(params services.XrayBuildParams) (*services.BuildScanResponse, error) {
+	buildScanService := services.NewBuildScanService(sm.client)
+	buildScanService.XrayDetails = sm.config.GetServiceDetails()
+	return buildScanService.GetBuildScanResults(params)
+}
+
+// BuildSummary will get the build summary from Xray
```
Comment (eyalbe4) [inline xray/manager.go:151]: BuildSummary will get the build summary from Xray
-->
BuildSummary returns the a summary of build scan which had been previously performed.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
I'd love to review this PR again following the upcoming changes,

File: xray/services/buildscan.go [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+	}
+	buildScanResponse := RequestBuildScanResponse{}
+	if err = json.Unmarshal(body, &buildScanResponse); err != nil {
+		return "", errorutils.CheckError(err)
+	}
+	buildScanInfo := buildScanResponse.Info
+	if strings.Contains(buildScanInfo, xrayScanBuildNotSelectedForIndexing) {
+		return "", errorutils.CheckErrorf(buildScanResponse.Info)
+	}
+	log.Info(buildScanInfo)
```
[the remark this answers is not in the document]
Comment (sverdlov93) [reply xray/services/buildscan.go:60]: it log from xray about the status of the scan

**Pull request merged**